lens-cube/src/main/java/org/apache/lens/cube/parse/CandidateUtil.java

```diff
@@ -0,0 +1,321 @@
+import com.google.common.collect.TreeRangeSet;
+
+import static org.apache.hadoop.hive.ql.parse.HiveParser.Identifier;
+
+/**
+ * Placeholder for Util methods that will be required for {@link Candidate}
+ */
+public class CandidateUtil {
+
+  /**
+   * Is calculated measure expression answerable by the Candidate
+   * @param exprNode
+   * @param candidate
+   * @param context
+   * @return
+   * @throws LensException
+   */
+  public static boolean isMeasureExpressionAnswerable(ASTNode exprNode, Candidate candidate, CubeQueryContext context)
+    throws LensException {
+    return candidate.getColumns().containsAll(HQLParser.getColsInExpr(
+      context.getAliasForTableName(context.getCube()), exprNode));
+  }
+
+  /**
+   * Returns true if the Candidate is valid for all the timeranges based on its start and end times.
+   * @param candidate
+   * @param timeRanges
+   * @return
+   */
+  public static boolean isValidForTimeRanges(Candidate candidate, List<TimeRange> timeRanges) {
+    for (TimeRange timeRange : timeRanges) {
+      if (!(timeRange.getFromDate().after(candidate.getStartTime())
+          && timeRange.getToDate().before(candidate.getEndTime()))) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  public static boolean isPartiallyValidForTimeRanges(Candidate cand, List<TimeRange> timeRanges) {
+    for (TimeRange timeRange : timeRanges) {
+      if ((cand.getStartTime().before(timeRange.getFromDate()) && cand.getEndTime().after(timeRange.getFromDate()))
+          || (cand.getStartTime().before(timeRange.getToDate()) && cand.getEndTime().after(timeRange.getToDate()))) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Gets the time partition columns for a storage candidate
+   * TODO decide is this needs to be supported for all Candidate types.
+   *
+   * @param candidate : Stoarge Candidate
+   * @param metastoreClient : Cube metastore client
+   * @return
+   * @throws LensException
+   */
+  public Set<String> getTimePartitionCols(StorageCandidate candidate, CubeMetastoreClient metastoreClient)
+    throws LensException {
+    Set<String> cubeTimeDimensions = candidate.getCube().getTimedDimensions();
+    Set<String> timePartDimensions = new HashSet<String>();
+    String singleStorageTable = candidate.getStorageName();
+    List<FieldSchema> partitionKeys = null;
+    partitionKeys = metastoreClient.getTable(singleStorageTable).getPartitionKeys();
+    for (FieldSchema fs : partitionKeys) {
+      if (cubeTimeDimensions.contains(CubeQueryContext.getTimeDimOfPartitionColumn(candidate.getCube(),
+        fs.getName()))) {
+        timePartDimensions.add(fs.getName());
+      }
+    }
+    return timePartDimensions;
+  }
+
+  /**
+   * Copy Query AST from sourceAst to targetAst
+   *
+   * @param sourceAst
+   * @param targetAst
+   * @throws LensException
+   */
+  public static void copyASTs(QueryAST sourceAst, QueryAST targetAst) throws LensException {
```

Comment:
Just a thought: Should this method be in `QueryAST` class seeing as it's doing all operations on the `targetAst`.